forms/app.py

```diff
@@ -1,379 +1,15 @@
-
-    if not IS_VALID_EMAIL(email):
-        if request_wants_json():
-            return jsonerror(400, {'error': "Invalid email address"})
-        else:
-            return render_template('error.html', 
-                                   title='Check email address', 
-                                   text='Email address %s is not formatted correctly' % str(email)), 400
-
-    # We're not using referrer anymore, just the domain + path
-    host = _referrer_to_path(flask.request.referrer)
-
-    if not host:
-        if request_wants_json():
-            return jsonerror(400, {'error': "Invalid \"Referrer\" header"})
-        else:
-            return render_template('error.html', 
-                                   title='Unable to submit form', 
-                                   text='Make sure your form is running on a proper server. For geeks: could not find the "Referrer" header.'), 400
-
-    # get the form for this request
-    form = Form.query.filter_by(hash=HASH(email, host)).first()
-
-    if form and form.confirmed:
-        return _send_form(form, email, host)
-
-    return _send_confirmation(form, email, host)
-
-
-def confirm_email(nonce):
-    ''' 
-    Confirmation emails point to this endpoint
-    It either rejects the confirmation or
-    flags associated email+host to be confirmed
-    '''
-
-    # get the form for this request
-    form = Form.query.filter_by(hash=nonce).first()
-
-    if not form:
-        return render_template('error.html', 
-                               title='Not a valid link', 
-                               text='Confirmation token not found.<br />Please check the link and try again.'), 400
-    
-    else:
-        form.confirmed = True
-        DB.session.add(form)
-        DB.session.commit()
-        return render_template('email_confirmed.html', email=form.email, host=form.host)
-
-
-def default(template='index'):
-    template = template if template.endswith('.html') else template+'.html'
-    return render_template(template, is_redirect = request.args.get('redirected'))
-
-
-def favicon():
-    return flask.redirect(url_for('static', filename='img/favicon.ico'))
-
-'''
-Add routes and create app (create_app is called in __init__.py)
-
-'''
-
-def configure_routes(app):
-    app.add_url_rule('/', 'index', view_func=default, methods=['GET'])
-    app.add_url_rule('/favicon.ico', view_func=favicon)
-    app.add_url_rule('/<email>', 'send', view_func=send, methods=['GET', 'POST'])
-    app.add_url_rule('/confirm/<nonce>', 'confirm_email', view_func=confirm_email, methods=['GET'])
-    app.add_url_rule('/thanks', 'thanks', view_func=thanks, methods=['GET'])
-    app.add_url_rule('/<path:template>', 'default', view_func=default, methods=['GET'])
-
-
-def create_app():
-    app = flask.Flask(__name__)
-    app.config.from_object(settings)
-
-    DB.init_app(app)
-    configure_routes(app)
```

Comment:
I was only using app to define routing, so no advantages as I can revert it and do it in other ways. If you prefer routing without decorators, I can also revert that.